pkg/kapp/app/recorded_app.go

```diff
@@ -187,6 +195,8 @@ func (a *RecordedApp) createOrUpdate(c *corev1.ConfigMap, labels map[string]stri
 		return fmt.Errorf("Creating app: %s", err)
 	}
 
+	a.setMeta(*configmap)
```

Comment:
Started setting `meta` here as in case of `--diff-run` we will unable to get the configmap later. In without `--diff-run` scenario it will reduce the `get` calls.